### PR TITLE
PROGRAMMES-6465 Update all HTTP APIs to use the Multi client

### PR DESCRIPTION
**Pull request by Tasiobg**
Update all HTTP APIs to use the Multi client and retire the single URL HttpApiClient

This is only for Electron, I'll open more PR for the remaining APIs and for removing the single HttpApiClient

- Updated Electron to use Multi client
- Fix missing ParseException in HttpApiMultiClient.php
- Fix MultiParseException, $responseKey can be integer or string
- Fix wrong use of MultiParseException in TupacService.php

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/ExternalApi/Exception/MultiParseException.php [hunk 1]
```diff
@@ -10,7 +10,7 @@ class MultiParseException extends \RuntimeException
     /** @var string */
     private $responseKey;
 
-    public function __construct(string $responseKey, $message = "", $code = 0, Exception $previous = null)
```
Comment (votemike) [inline src/ExternalApi/Exception/MultiParseException.php]: Why was this removed?

Comment (Tasiobg) [reply src/ExternalApi/Exception/MultiParseException.php]: We found an issue because `$responseKey` can be integer or string

Comment (votemike) [reply src/ExternalApi/Exception/MultiParseException.php]: Brilliant.... OK.

Comment (votemike) [reply src/ExternalApi/Exception/MultiParseException.php]: Should you update the docblock then?

File: src/ExternalApi/Tupac/Service/TupacService.php [hunk 2]
```diff
@@ -63,13 +63,13 @@ private function generateRequestUrl(array $recordIds, bool $isUk = false): strin
     private function parseResponse(array $responses): array
     {
         $results = [];
-        foreach ($responses as $response) {
+        foreach ($responses as $key => $response) {
             if (!$response instanceof Response) {
                 throw new MultiParseException("TUPAC callback received non-response object!");
```
Comment (BjornTwachtmann) [inline src/ExternalApi/Tupac/Service/TupacService.php]: You missed the response key here, no?
